server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/http/UserCredentialParserTest.scala

```diff
@@ -0,0 +1,127 @@
+ * ***************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.nio.charset.StandardCharsets
+import java.util.Base64
+
+import eu.timepit.refined.auto._
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class UserCredentialParserTest {
+  @Test
+  def shouldReturnCredentialsWhenUsernamePasswordToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1:password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential("user1", "password")))
+  }
+
+  @Test
+  def shouldReturnCredentialsWhenRandomSpecialCharacterInUsernameToken(): Unit = {
+    val token: String = "Basic " + toBase64("fd2*#jk:password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential("fd2*#jk", "password")))
+  }
+
+  @Test
+  def shouldReturnCredentialsWhenRandomSpecialCharacterInBothUsernamePasswoedToken(): Unit = {
+    val token: String = "Basic " + toBase64("fd2*#jk:password@fd23*&^$%")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential("fd2*#jk", "password@fd23*&^$%")))
+  }
+
+  @Test
+  def shouldReturnCredentialsWhenUsernameDomainPasswordToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1@domain.tld:password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential("user1@domain.tld", "password")))
+  }
+
+  @Test
+  def shouldReturnNoneWhenPayloadIsNotBase64(): Unit = {
+    val token: String = "Basic user1:password"
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldThrowWhenInvalidToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1@password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldReturnEmptyWhenWithEmptyToken(): Unit = {
+    assertThat(UserCredential.parseUserCredentials(""))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldThrowWhenWithWrongFormatToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1@password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+       .isEqualTo(None)
+  }
+
+  @Test
+  def shouldReturnEmptyWhenWithUpperCaseToken(): Unit = {
+    val token: String = "BASIC " + toBase64("user1@password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldReturnEmptyWhenWithLowerCaseToken(): Unit = {
+    val token: String = "basic " + toBase64("user1:password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldReturnEmptyWhenNoPassword(): Unit = {
+    val token: String = "Basic " + toBase64("user1:")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(None)
```

Comment:
Well technically UserCredential::parseUserCredentials is not the right place to enforce password not to be empty. I propose we support that case.